src/Vehicle/CompInfo.h

```diff
@@ -27,17 +28,43 @@ class CompInfo : public QObject
 public:
     CompInfo(COMP_METADATA_TYPE type, uint8_t compId, Vehicle* vehicle, QObject* parent = nullptr);
 
-    /// Called to pass the COMPONENT_INFORMATION message in
-    void setMessage(const mavlink_message_t& message);
+    const QString& uriMetaData() const { return _uris.uriMetaData; }
+    const QString& uriTranslation() const { return _uris.uriTranslation; }
+
+    uint32_t crcMetaData() const { return _uris.crcMetaData; }
+    bool crcMetaDataValid() const { return _uris.crcMetaDataValid; }
+    uint32_t crcTranslation() const { return _uris.crcTranslation; }
+    bool crcTranslationValid() const { return _uris.crcTranslationValid; }
+
+    void setUriMetaData(const QString& uri, uint32_t crc);
 
     virtual void setJson(const QString& metaDataJsonFileName, const QString& translationJsonFileName) = 0;
 
-    COMP_METADATA_TYPE  type;
-    Vehicle*            vehicle            = nullptr;
-    uint8_t             compId             = MAV_COMP_ID_ALL;
-    bool                available          = false;
-    uint32_t            uidMetaData        = 0;
-    uint32_t            uidTranslation     = 0;
-    QString             uriMetaData;
-    QString             uriTranslation;
+    bool available() const { return !_uris.uriMetaData.isEmpty(); }
+
+    const COMP_METADATA_TYPE  type;
+    Vehicle* const      vehicle                = nullptr;
+    const uint8_t       compId                 = MAV_COMP_ID_ALL;
+
+private:
+    friend class CompInfoGeneral;
+
+    struct Uris {
+        bool                crcMetaDataValid            = false;
```

Comment:
I don't get these `Valid` flags? Why isn't a CRC always required?

Reply:
For translation (and potentially fallback) there's no CRC, as the autopilot does not see that file. So I added the flag to all, but yes, for the main metadata it needs to be valid and this one could be removed.